enterprise/tools/replay_action/replay_action.go

```diff
@@ -790,6 +872,27 @@ func (r *Replayer) execute(ctx, srcCtx, targetCtx context.Context, action *repb.
 
 		sourceExecutionRN = digest.NewCASResourceName(ad, *targetRemoteInstanceName, sourceExecutionRN.GetDigestFunction())
 	}
+	// If we're running multiple executions of the action, and the do_not_cache
+	// field isn't set, then the action merger is likely to dedupe some of the
+	// executions. So we need to override the do_not_cache field to true in
```

Comment:
If we're only running 1 execution, there are still some edge cases where it could be merged, right? Is there a reason to not just always set `do_not_cache`?

Reply:
I think it would also be okay to add a header to disable action merging so that we don't need to modify the action.

Reply:
> If we're only running 1 execution, there are still some edge cases where it could be merged, right? Is there a reason to not just always set do_not_cache?

Probably, but I'm not sure I've hit this in practice yet.

> I think it would also be okay to add a header to disable action merging so that we don't need to modify the action.

The thought also crossed my mind of doing it this way - wasn't sure if folks would be cool with it since idk of any use cases outside of this tool. I can send that as a follow-up since it simplifies things quite a bit.